### ISSUE TITLE
training simclr on multi-gpu

### ISSUE THREAD

**Issue by bhat-prashant** [google-research/simclr#102]
Hi, 
I am really new to self supervised learning. I would like to know how do you adjust no.of epochs when you train on multi-gpu. 

I am intending to train on ImageNet. Unfortunately, I cannot fit mini-batch of 256 in one gpu. As for as I understand, when I train on 4-gpu, even if I train for 100 epochs, the no.of epochs trained will be effectively 100/4=25 as number of gradient updates will also be divided by 4. Please correct me if I am wrong. 

Please let me know how have you accounted for that in your implementation. 

Thanks

**Comment by chentingpc** [comment]
Two concepts here: (1) number of epochs, that's the number of passes over whole dataset; (2) number of grad/updating steps, that is determined by batch size as well as number of epochs. None of them has anything to do with multi-gpu. I guess what you meant is when increasing batch size for the same epochs, the grad updates will be fewer; which is true, but you can also increase learning rate with batch size so that within a certain size of batch (search linear scaling or sqrt scaling in the paper), you can always get the same performance in the same number of epochs.
